modules/core/src/scenegraph/group.js

```diff
@@ -47,7 +47,7 @@ export default class Group extends Node {
       if (child instanceof Group) {
         result = child.traverse(visitor, {modelMatrix});
       } else {
-        // child.setUniforms({modelMatrix});
+        child.setMatrix(modelMatrix); // TODO: can child has his own matrix?
```

Comment:
Yes every scenegraph node has its own matrix. What we need to do is to multiply them together as we descend and pass the current matrix to `traverse(child, parentMatrix)`